src/test/ui/moves/move-deref-coercion.stderr

```diff
@@ -0,0 +1,35 @@
+error[E0382]: borrow of moved value: `val`
+  --> $DIR/move-deref-coercion.rs:25:5
+   |
+LL |     let _val = val.first;
+   |                --------- value moved here
+LL |     val.inner;
+   |     ^^^^^^^^^ value borrowed here after partial move
+   |
+   = note: borrow occurs due to deref coercion to `NotCopy`
+note: deref defined here
+  --> $DIR/move-deref-coercion.rs:17:5
+   |
+LL |     type Target = NotCopy;
+   |     ^^^^^^^^^^^^^^^^^^^^^^
+   = note: move occurs because `val.first` has type `NotCopy`, which does not implement the `Copy` trait
```

Comment:
This `note` seems like it could go as the label of the main span in the previous `note`.

There's something missing in the new output, but I can't quite put my finger on it. Having said that, we can land this PR and improve it later.

Reply:
I don't think we should attach it to the span of `val.inner;`, since that's referring to the borrow. However, I think it would make sense to reorder it to go before the 'deref defined here' note.